# EDA Datasets Youtube

## Datasets Yang Digunakan
</br></br>
Datasets yang digunakan adalah datasets video Youtube yang diambil dari Kaggle dengan link sebagai berikut : <br>https://www.kaggle.com/datasnaek/youtube-new


Datasets yang digunakan hanyalah data dari region CA, GB, dan US.


Jenis dari datasets dari ketiga region tersebut adalah sama yaitu : <b>
 - Data video berbentuk .csv <br>
 - Data kategori berbentuk .json</b>

## Import library dan load datasets
Import library <b>Pandas</b> dan library lain yang diperlukan untuk membaca datasets .csv dan .json. <br/><br/>
Datasets yang diload adalah sebagai berikut : <b>
- CA_category_id.json
- CAvideos.csv
- GB_category_id.json
- GBvideos.csv
- US_category_id.json
- USvideos.csv </b>

Datasets tersebut diubah menjadi dataframe terlebih dahulu dengan library Pandas agar dapat diolah.

In [1]:
#Import library
import pandas as pd
import json as js
from pandas.io.json import json_normalize

#load datasets CA
dfCAvideos = pd.read_csv("datasets/CAvideos.csv")
with open("datasets/CA_category_id.json") as file:    
    jsCAcategory = js.load(file)

#Load datasets GB
dfGBvideos = pd.read_csv("datasets/GBvideos.csv")
with open("datasets/GB_category_id.json") as file:    
    jsGBcategory = js.load(file)
    
#Load datasets US
dfUSvideos = pd.read_csv("datasets/USvideos.csv")
with open("datasets/US_category_id.json") as file:    
    jsUScategory = js.load(file)

# set option agar float dapat ditampilkan seluruhnya
pd.set_option('float_format', '{:f}'.format)

## Dataset Video
Dataset videos adalah dataset video YouTube di yang terdiri dari kolom sebagai berikut :
 - <b>video_id</b> (ID Video) </br>
 - <b>trending_date</b> (Tanggal Trending) </br>
 - <b>title</b> (Judul Video) </br>
 - <b>channel_title</b> (Nama Channel)</br>
 - <b>category_id</b> (ID Kategori) </br>
 - <b>publish_time</b> (Tanggal Upload) <br>
 - <b>tags</b> (Tag-tag) </br>
 - <b>views</b> (Jumlah Penonton) </br>
 - <b>likes</b> (Jumlah yang Menyukai Video) </br>
 - <b>dislikes</b> (Jumlah yang Tidak Menyukai Video) </br>
 - <b>comment_count</b> (Jumlah Komentar) </br>
 - <b>thumbnail_link</b> (Link Thumbnail) </br>
 - <b>comments_disabled</b> (Status Kolom Komentar) </br>
 - <b>ratings_disabled</b> (Status Rating) </br>
 - <b>video_error_or_removed</b> (Status Video Error/Dihapus) </br>
 - <b>description</b> (Deskripsi Video)
 
Dataset video dari 3 region adalah sebagai berikut :
- CAvideos.csv (dataset video region Canada)
- GBvideos.csv (dataset video region Great Britain)
- USvideos.csv (dataset video region United States)

Karena ke-3nya berbentuk .csv maka dapat diubah menjadi dataframe dengan sintaks pd.read_csv seperti yang dilakukan di cell sebelumnya.

### Data Video Region Canada
Berikut adalah data video dari region Canada :

In [2]:
dfCAvideos

video_id trending_date  \
0      n1WpP7iowLc      17.14.11   
1      0dBIkQ4Mz1M      17.14.11   
2      5qpjK5DgCt4      17.14.11   
3      d380meD0W0M      17.14.11   
4      2Vv-BfVoq4g      17.14.11   
5      0yIWz1XEeyc      17.14.11   
6      _uM5kFfkhB8      17.14.11   
7      2kyS6SvSYSE      17.14.11   
8      JzCsM1vtn78      17.14.11   
9      43sm-QwLcx4      17.14.11   
10     H1KBHFXm2Bg      17.14.11   
11     U3xLOo-CNwo      17.14.11   
12     FyZMnhUtLfE      17.14.11   
13     7MxiQ4v0EnE      17.14.11   
14     LUzsOyWp9lw      17.14.11   
15     AS9-ITLhQxo      17.14.11   
16     gifPYwArCVQ      17.14.11   
17     8NHA23f7LvU      17.14.11   
18     fy-CuCzaPp8      17.14.11   
19     kzwfHumJyYc      17.14.11   
20     nc99ccSXST0      17.14.11   
21     puqaWrEC7tY      17.14.11   
22     6H3wVsceoS0      17.14.11   
23     PRlmRnU9qts      17.14.11   
24     ZhhXLMbZ1rQ      17.14.11   
25     wuf2KinLxZ8      17.14.11   
26     aVTAU_4i9AY      17.14.11   
27     7779JdxVAg0      17.14.11   
28     jVpp2BUtMoY      17.14.11   
29     j67FgNEvyh8      17.14.11   
...            ...           ...   
40851  0iuAf4--B6g      18.14.06   
40852  pP9fQZsstIg      18.14.06   
40853  U-9tUEOFKNU      18.14.06   
40854  vaPeQE9m4_U      18.14.06   
40855  yiAbt_u3DAE      18.14.06   
40856  3HuxnX6hoD0      18.14.06   
40857  vg0ZfeszGrU      18.14.06   
40858  QqMUz2gj7H8      18.14.06   
40859  ccYF1Tv301g      18.14.06   
40860  i7ukVwJur_Q      18.14.06   
40861  JGm9Y_hFqNk      18.14.06   
40862  VI4NZW8q6nU      18.14.06   
40863  Tmv5f02jA5U      18.14.06   
40864  Fmdb-KmlzD8      18.14.06   
40865  0-VAnh7r-_8      18.14.06   
40866  YOqj7JtYPa8      18.14.06   
40867  xxbkfmYFxnw      18.14.06   
40868  9gkxwIZPOZk      18.14.06   
40869  sbxP1ndw2ds      18.14.06   
40870  6Mf2ylffKws      18.14.06   
40871  FoOi-oTaIEU      18.14.06   
40872  uTGsMan-6Ss      18.14.06   
40873  i1Zym3v8tL4      18.14.06   
40874  4uO08Lr5g3k      18.14.06   
40875  7E1np354AEc      18.14.06   
40876  sGolxsMSGfQ      18.14.06   
40877  8HNuRNi8t70      18.14.06   
40878  GWlKEM3m2EE      18.14.06   
40879  lbMKLzQ4cNQ      18.14.06   
40880  POTgw38-m58      18.14.06   

                                                   title  \
0             Eminem - Walk On Water (Audio) ft. Beyoncé   
1                          PLUSH - Bad Unboxing Fan Mail   
2      Racist Superman | Rudy Mancuso, King Bach & Le...   
3                               I Dare You: GOING BALD!?   
4            Ed Sheeran - Perfect (Official Music Video)   
5      Jake Paul Says Alissa Violet CHEATED with LOGA...   
6                 Vanoss Superhero School - New Students   
7                     WE WANT TO TALK ABOUT OUR MARRIAGE   
8                   THE LOGANG MADE HISTORY. LOL. AGAIN.   
9      Finally Sheldon is winning an argument about t...   
10       21 Savage - Bank Account (Official Music Video)   
11     12 Weird Ways To Sneak Food Into Class / Back ...   
12            猎场 | Game Of Hunting 12【TV版】（胡歌、張嘉譯、祖峰等主演）   
13     Daang ( Full Video ) | Mankirt Aulakh | Sukh S...   
14     YOUTUBERS REACT TO TOP 10 TWITTER ACCOUNTS OF ...   
15               I Hired An MI6 Spy To Help Me Disappear   
16                        Fake Pet Smart Employee Prank!   
17     Jason Momoa Wows Hugh Grant With Some Dothraki...   
18     Rooster Teeth Animated Adventures - Drunk Baby...   
19     Sharry Mann: Cute Munda ( Song Teaser) | Parmi...   
20                   5 Ice Cream Gadgets put to the Test   
21                      Nickelback Lyrics: Real or Fake?   
22         Smart Girl Shows How to Build Candy Dispenser   
23     Jake Paul - Saturday Night (Song) feat. Nick C...   
24     DO COLLEGE KIDS KNOW 80s MUSIC? #8 (REACT: Do ...   
25                     WRESTLING NIGAHIGA!!! (WILD CARD)   
26                           Throwing Things Into A Fan!   
27     10 FILMS que LES PARENTS CROYAIENT ÊTRE pour E...   
28                               So W

In [3]:
dfCAvideos.shape

(40881, 16)

Berdasarkan pengecekan data video dari region Canada memiliki dimensi <b>40881 baris x 16 kolom</b>.

### Data Video Region Great Britain
Berikut adalah data video dari region Great Britain :

In [4]:
dfGBvideos

video_id trending_date  \
0      Jw1Y-zhQURU      17.14.11   
1      3s1rvMFUweQ      17.14.11   
2      n1WpP7iowLc      17.14.11   
3      PUTEiSjKwJU      17.14.11   
4      rHwDegptbI4      17.14.11   
5      AumaWl0TNBo      17.14.11   
6      2Vv-BfVoq4g      17.14.11   
7      -N5eucPMTTc      17.14.11   
8      LMCuKltaY3M      17.14.11   
9      9t9u_yPEidY      17.14.11   
10     ONQ-fAp5X64      17.14.11   
11     Dlwf5u2BLAg      17.14.11   
12     NVCqz2BDfcQ      17.14.11   
13     MyEqfBkzESU      17.14.11   
14     b4vTZx_AtHk      17.14.11   
15     BsfhHKx6ajA      17.14.11   
16     DIU3xPdhCBI      17.14.11   
17     pz95u3UVpaM      17.14.11   
18     jp9hK-jY6yY      17.14.11   
19     gEHCXl4J9Qo      17.14.11   
20     OWvCCu2t6Ts      17.14.11   
21     6p-QzY5bxJ0      17.14.11   
22     fiusxyygqGk      17.14.11   
23     QOksZ8VogRw      17.14.11   
24     lo5ERI5nPbM      17.14.11   
25     Ja_GMU7-sjs      17.14.11   
26     _6ViZ13mN9I      17.14.11   
27     F62QpaPWXDI      17.14.11   
28     g5c1bk8weaQ      17.14.11   
29     3qYt_fXdaXY      17.14.11   
...            ...           ...   
38886  iaL3EEWspLo      18.14.06   
38887  14G7Qw5DC9M      18.14.06   
38888  bGW2eZGnznk      18.14.06   
38889  nkWn5vKCZj4      18.14.06   
38890  uBYzYji4rmE      18.14.06   
38891  niviooakBLs      18.14.06   
38892  3rkXFyt7EXw      18.14.06   
38893  BgR_HJ7aGDY      18.14.06   
38894  N7HFXgyTya0      18.14.06   
38895  m6lY1GXTu5M      18.14.06   
38896  71Es-8FfATo      18.14.06   
38897  pFc6I0rgmgY      18.14.06   
38898  gf7FV1i7lb8      18.14.06   
38899  osGl1krYP-g      18.14.06   
38900  QW8whgmyTNU      18.14.06   
38901  BspHjvU11y4      18.14.06   
38902  mzB1VGEGcSU      18.14.06   
38903  igT1QKKm2GE      18.14.06   
38904  9_LIP7qguYw      18.14.06   
38905  8h--kFui1JA      18.14.06   
38906  2PH7dK6SLC8      18.14.06   
38907  HbaQ9xnoMGQ      18.14.06   
38908  Un8b0JDw0Bc      18.14.06   
38909  Ibp6KqGRVbY      18.14.06   
38910  uKxWP56VStM      18.14.06   
38911  l884wKofd54      18.14.06   
38912  IP8k2xkhOdI      18.14.06   
38913  Il-an3K9pjg      18.14.06   
38914  -DRsfNObKIQ      18.14.06   
38915  4YFo4bdMO8Q      18.14.06   

                                                   title  \
0          John Lewis Christmas Ad 2017 - #MozTheMonster   
1              Taylor Swift: …Ready for It? (Live) - SNL   
2             Eminem - Walk On Water (Audio) ft. Beyoncé   
3      Goals from Salford City vs Class of 92 and Fri...   
4      Dashcam captures truck's near miss with child ...   
5                           How My Relationship Started!   
6            Ed Sheeran - Perfect (Official Music Video)   
7                          CHRISTMAS HAS GONE TO MY HEAD   
8       Elbow - Golden Slumbers (John Lewis Advert 2017)   
9      Jennifer Lopez - Amor, Amor, Amor (Official Vi...   
10           CAN BABIES DO GYMNASTICS?  **World Record**   
11     TV Reporter Interviews Alleged Bank Robber, Th...   
12                  Gucci Mane  - Curve (ft. The Weeknd)   
13                     EXTREME GOLDEN TAN TRANSFORMATION   
14                      How To Dry a Shirt in 30 Seconds   
15     Attracting An Alpaca With An Accordion | Nowhe...   
16     Sam Smith surprises brides at their wedding! (...   
17     Camila Cabello - Havana (Vertical Video) ft. Y...   
18         When Someone Has A Crush on You | Lilly Singh   
19       Taylor Swift - “New Year’s Day” Fan Performance   
20     SO Much To Tell You | 32 Weeks Pregnant | MOTH...   
21     The 'Stranger Things' Kids Were Nearly a Motow...   
22          Marshmello - You & Me (Official Music Video)   
23     Bastille - World Gone Mad (from Bright: The Al...   
24       Sharon Needles - Andy Warhol Is Dead [Official]   
25                     Google Pixelbook: $1000 of Weird!   
26     GSP and Tyron Woodley Talk About Their Potenti...   
27                       THE CHEAPEST & BEST EYESHADOWS?   
28            FIRST TIME IM DOING THI

In [5]:
dfGBvideos.shape

(38916, 16)

Berdasarkan pengecekan data video dari region Great Britain memiliki dimensi <b>38916 baris x 16 kolom</b>.

### Data Video Region Great Britain
Berikut adalah data video dari region Great Britain :

In [6]:
dfUSvideos

video_id trending_date  \
0      2kyS6SvSYSE      17.14.11   
1      1ZAPwfrtAFY      17.14.11   
2      5qpjK5DgCt4      17.14.11   
3      puqaWrEC7tY      17.14.11   
4      d380meD0W0M      17.14.11   
5      gHZ1Qz0KiKM      17.14.11   
6      39idVpFF7NQ      17.14.11   
7      nc99ccSXST0      17.14.11   
8      jr9QtXwC9vc      17.14.11   
9      TUmyygCMMGA      17.14.11   
10     9wRQljFNDW8      17.14.11   
11     VifQlJit6A0      17.14.11   
12     5E4ZBSInqUU      17.14.11   
13     GgVmn66oK_A      17.14.11   
14     TaTleo4cOs8      17.14.11   
15     kgaO45SyaO4      17.14.11   
16     ZAQs-ctOqXQ      17.14.11   
17     YVfyYrEmzgM      17.14.11   
18     eNSN6qet1kE      17.14.11   
19     B5HORANmzHw      17.14.11   
20     vU14JY3x81A      17.14.11   
21     6VhU_T463sU      17.14.11   
22     _-aDHxoblr4      17.14.11   
23     JBZTZZAcFTw      17.14.11   
24     lZ68j2J_GOM      17.14.11   
25     dRpNZV18N_g      17.14.11   
26     fcVjitaM3LY      17.14.11   
27     qeWvgZLz9yU      17.14.11   
28     iIxy3JN3-jc      17.14.11   
29     n30k5CwLhS4      17.14.11   
...            ...           ...   
40919  vDyYMcjf500      18.14.06   
40920  3Q2j5ApzSqs      18.14.06   
40921  GfxNGjfSKRY      18.14.06   
40922  amtC28yfYCM      18.14.06   
40923  9NC2saE5MbQ      18.14.06   
40924  XdNOI-q70q4      18.14.06   
40925  QgOXIEhHU1Y      18.14.06   
40926  SQsPvrev_bQ      18.14.06   
40927  _1LdMWlNYS4      18.14.06   
40928  AFbu21AGSho      18.14.06   
40929  hHaUQh6Gx-Q      18.14.06   
40930  wI89nVn6LHk      18.14.06   
40931  oLDbO545aKQ      18.14.06   
40932  tiF9SzzzX_o      18.14.06   
40933  Gs26bZTRkdU      18.14.06   
40934  Qcj15vHJTtk      18.14.06   
40935  2in8XqiElwc      18.14.06   
40936  Xr2rgT9uEnA      18.14.06   
40937  v7H_Or9Nr5I      18.14.06   
40938  n_W54baizX8      18.14.06   
40939  pcJo0tIWybY      18.14.06   
40940  _QWZvU7VCn8      18.14.06   
40941  7UoP9ABJXGE      18.14.06   
40942  ju_inUnrLc4      18.14.06   
40943  1PhPYr_9zRY      18.14.06   
40944  BZt0qjTWNhw      18.14.06   
40945  1h7KV2sjUWY      18.14.06   
40946  D6Oy4LfoqsU      18.14.06   
40947  oV0zkMe1K8s      18.14.06   
40948  ooyjaVdt-jA      18.14.06   

                                                   title  \
0                     WE WANT TO TALK ABOUT OUR MARRIAGE   
1      The Trump Presidency: Last Week Tonight with J...   
2      Racist Superman | Rudy Mancuso, King Bach & Le...   
3                       Nickelback Lyrics: Real or Fake?   
4                               I Dare You: GOING BALD!?   
5                                  2 Weeks with iPhone X   
6              Roy Moore & Jeff Sessions Cold Open - SNL   
7                    5 Ice Cream Gadgets put to the Test   
8      The Greatest Showman | Official Trailer 2 [HD]...   
9      Why the rise of the robots won’t mean the end ...   
10     Dion Lewis' 103-Yd Kick Return TD vs. Denver! ...   
11     (SPOILERS) 'Shiva Saves the Day' Talked About ...   
12            Marshmello - Blocks (Official Music Video)   
13                Which Countries Are About To Collapse?   
14                              SHOPPING FOR NEW FISH!!!   
15                                      The New SpotMini   
16      One Change That Would Make Pacific Rim a Classic   
17     How does your body know you're full? - Hilary ...   
18                            HomeMade Electric Airplane   
19              Founding An Inbreeding-Free Space Colony   
20                      How Can You Control Your Dreams?   
21     The Making of Hela's Headdress from Thor: Ragn...   
22     Is It Dangerous To Talk To A Camera While Driv...   
23     What $4,800 Will Get You In NYC | Sweet Digs H...   
24                          Using Other People's Showers   
25                SPAGHETTI BURRITO VS SPAGHETTI BURRITO   
26                  78557 and Proth Primes - Numberphile   
27       A Smart... MUG?! - Take apart a Heated Thermos!   
28     LeBron James admits he was rip

In [7]:
dfUSvideos.shape

(40949, 16)

Berdasarkan pengecekan data video dari region United States memiliki dimensi <b>40849 baris x 16 kolom</b>.

<b>------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------</b>

<b>------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------</b>

## Dataset Kategori
Dataset kategori adalah dataset kategori video di Youtube.

Dataset kategori dari 3 region tersebut adalah :
- CA_category_id.json (dataset kategori video region Canada)
- GB_category_id.json (dataset kategori video region Great Britain)
- US_category_id.json (dataset kategori video region United States)

Karena dataset berbentuk .json maka perlu dilakukan transformasi untuk mengubahnya menjadi dataframe.

Mari melihat isi dari file kategori .json dari region Canada sebagai contoh.

In [8]:
jsCAcategory

{'kind': 'youtube#videoCategoryListResponse',
 'etag': '"ld9biNPKjAjgjV7EZ4EKeEGrhao/1v2mrzYSYG6onNLt2qTj13hkQZk"',
 'items': [{'kind': 'youtube#videoCategory',
   'etag': '"ld9biNPKjAjgjV7EZ4EKeEGrhao/Xy1mB4_yLrHy_BmKmPBggty2mZQ"',
   'id': '1',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
    'title': 'Film & Animation',
    'assignable': True}},
  {'kind': 'youtube#videoCategory',
   'etag': '"ld9biNPKjAjgjV7EZ4EKeEGrhao/UZ1oLIIz2dxIhO45ZTFR3a3NyTA"',
   'id': '2',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
    'title': 'Autos & Vehicles',
    'assignable': True}},
  {'kind': 'youtube#videoCategory',
   'etag': '"ld9biNPKjAjgjV7EZ4EKeEGrhao/nqRIq97-xe5XRZTxbknKFVe5Lmg"',
   'id': '10',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
    'title': 'Music',
    'assignable': True}},
  {'kind': 'youtube#videoCategory',
   'etag': '"ld9biNPKjAjgjV7EZ4EKeEGrhao/HwXKamM1Q20q9BN-oBJavSGkfDI"',
   'id': '15',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdnt

Berdasarkan pengecekan tersebut, dataset kategori memiliki susunan sebagai berikut :
 - kind<br>
 - etag<br>
 - id<br>
 - items<br>
 - snippet<br>
 - title<br>
 - assignable<br>

Kemudian ditemukan bahwa items adalah suatu list yang berisi kind, etag, id, snippet, title dan assignable.

### Mentransformasi data kategori JSON menjadi dataframe
Untuk keperluan analisis, yang dibutuhkan di sini adalah id kategori dan title / nama kategori. Maka 2 hal tersebut yang akan diekstrak kemudian dimasukkan ke dalam dataframe.

#### Normalisasi list items di data kategori
Karena yang dibutuhkan adalah id dan title di dalam list items, maka items harus dinormalisasi dengan menggunakan sintaks <b>json_normalize()</b>.
Dengan json_normalize, Pandas dapat mengambil objek JSON yang bersarang dan mengubahnya menjadi dataframe.

In [9]:
jsCAcatitem = json_normalize(jsCAcategory['items'])
jsGBcatitem = json_normalize(jsGBcategory['items'])
jsUScatitem = json_normalize(jsUScategory['items'])

Mari ambil contoh items yang telah dinormalisasi dari data kategori region Canada

In [10]:
jsCAcatitem

etag  id  \
0   "ld9biNPKjAjgjV7EZ4EKeEGrhao/Xy1mB4_yLrHy_BmKm...   1   
1   "ld9biNPKjAjgjV7EZ4EKeEGrhao/UZ1oLIIz2dxIhO45Z...   2   
2   "ld9biNPKjAjgjV7EZ4EKeEGrhao/nqRIq97-xe5XRZTxb...  10   
3   "ld9biNPKjAjgjV7EZ4EKeEGrhao/HwXKamM1Q20q9BN-o...  15   
4   "ld9biNPKjAjgjV7EZ4EKeEGrhao/9GQMSRjrZdHeb1OEM...  17   
5   "ld9biNPKjAjgjV7EZ4EKeEGrhao/FJwVpGCVZ1yiJrqZb...  18   
6   "ld9biNPKjAjgjV7EZ4EKeEGrhao/M-3iD9dwK7YJCafRf...  19   
7   "ld9biNPKjAjgjV7EZ4EKeEGrhao/WmA0qYEfjWsAoyJFS...  20   
8   "ld9biNPKjAjgjV7EZ4EKeEGrhao/EapFaGYG7K0StIXVf...  21   
9   "ld9biNPKjAjgjV7EZ4EKeEGrhao/xId8RX7vRN8rqkbYZ...  22   
10  "ld9biNPKjAjgjV7EZ4EKeEGrhao/G9LHzQmx44rX2S5ya...  23   
11  "ld9biNPKjAjgjV7EZ4EKeEGrhao/UVB9oxX2Bvqa_w_y3...  24   
12  "ld9biNPKjAjgjV7EZ4EKeEGrhao/QiLK0ZIrFoORdk_g2...  25   
13  "ld9biNPKjAjgjV7EZ4EKeEGrhao/r6Ck6Z0_L0rG37VJQ...  26   
14  "ld9biNPKjAjgjV7EZ4EKeEGrhao/EoYkczo9I3RCf96Rv...  27   
15  "ld9biNPKjAjgjV7EZ4EKeEGrhao/w5HjcTD82G_XA3xBc...  28   
16  "ld9biNPKjAjgjV7EZ4EKeEGrhao/lL7uWDr_071CHxifj...  30   
17  "ld9biNPKjAjgjV7EZ4EKeEGrhao/WnuVfjO-PyFLO7NTR...  31   
18  "ld9biNPKjAjgjV7EZ4EKeEGrhao/ctpH2hGA_UZ3volJT...  32   
19  "ld9biNPKjAjgjV7EZ4EKeEGrhao/L0kR3-g1BAo5UD1PL...  33   
20  "ld9biNPKjAjgjV7EZ4EKeEGrhao/pUZOAC_s9sfiwar63...  34   
21  "ld9biNPKjAjgjV7EZ4EKeEGrhao/Xb5JLhtyNRN3AQq02...  35   
22  "ld9biNPKjAjgjV7EZ4EKeEGrhao/u8WXzF4HIhtEi805_...  36   
23  "ld9biNPKjAjgjV7EZ4EKeEGrhao/D04PP4Gr7wc4IV_O9...  37   
24  "ld9biNPKjAjgjV7EZ4EKeEGrhao/i5-_AceGXQCEEMWU0...  38   
25  "ld9biNPKjAjgjV7EZ4EKeEGrhao/rtlxd0zOixA9QHdIZ...  39   
26  "ld9biNPKjAjgjV7EZ4EKeEGrhao/N1TrDFLRppxZgBowC...  40   
27  "ld9biNPKjAjgjV7EZ4EKeEGrhao/7UMGi6zRySqXopr_r...  41   
28  "ld9biNPKjAjgjV7EZ4EKeEGrhao/RScXhi324h8usyIet...  42   
29  "ld9biNPKjAjgjV7EZ4EKeEGrhao/0n9MJVCDLpA8q7aiG...  43   
30  "ld9biNPKjAjgjV7EZ4EKeEGrhao/x5NxSf5fz8hn4loSN...  44   

                     kind  snippet.assignable         snippet.channelId  \
0   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
1   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
2   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
3   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
4   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
5   youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
6   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
7   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
8   youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
9   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
10  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
11  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
12  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
13  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
14  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
15  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
16  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
17  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
18  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
19  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
20  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
21  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
22  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
23  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
24  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
25  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
26  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ

#### Memasukkan isi dari items ke dalam dataframe
items dari semua dataset kategori dari ke-3 region dimasukkan menjadi dataframe agar mudah untuk dikelola

In [11]:
dfCAcategory = jsCAcatitem
dfGBcategory = jsGBcatitem
dfUScategory = jsUScatitem

### Mentransformasi Data kategori Menjadi ID dan Nama Kategori Saja
ID dan nama category/snippet title dibutuhkan untuk dimerge dengan dataframe video. Maka harus dilakukan drop pada kolom yang tidak diperlukan pada dataframe.

In [12]:
dfCAcategory1 = dfCAcategory.drop(['etag','kind','snippet.assignable','snippet.channelId'], axis=1)
dfGBcategory1 = dfGBcategory.drop(['etag','kind','snippet.assignable','snippet.channelId'], axis=1)
dfUScategory1 = dfUScategory.drop(['etag','kind','snippet.assignable','snippet.channelId'], axis=1)

In [13]:
dfCAcategory1

id         snippet.title
0    1      Film & Animation
1    2      Autos & Vehicles
2   10                 Music
3   15        Pets & Animals
4   17                Sports
5   18          Short Movies
6   19       Travel & Events
7   20                Gaming
8   21         Videoblogging
9   22        People & Blogs
10  23                Comedy
11  24         Entertainment
12  25       News & Politics
13  26         Howto & Style
14  27             Education
15  28  Science & Technology
16  30                Movies
17  31       Anime/Animation
18  32      Action/Adventure
19  33              Classics
20  34                Comedy
21  35           Documentary
22  36                 Drama
23  37                Family
24  38               Foreign
25  39                Horror
26  40        Sci-Fi/Fantasy
27  41              Thriller
28  42                Shorts
29  43                 Shows
30  44              Trailers

kolom snippet.title direname menjadi title agar tidak membingungkan.

In [14]:
dfCAcategory1.rename(columns={'snippet.title': 'title'}, inplace=True)
dfGBcategory1.rename(columns={'snippet.title': 'title'}, inplace=True)
dfUScategory1.rename(columns={'snippet.title': 'title'}, inplace=True)

### Data Kategori Video Region Canada
Berikut adalah data kategori video region Canada setelah diubah menjadi dataframe :

In [15]:
dfCAcategory1

id                 title
0    1      Film & Animation
1    2      Autos & Vehicles
2   10                 Music
3   15        Pets & Animals
4   17                Sports
5   18          Short Movies
6   19       Travel & Events
7   20                Gaming
8   21         Videoblogging
9   22        People & Blogs
10  23                Comedy
11  24         Entertainment
12  25       News & Politics
13  26         Howto & Style
14  27             Education
15  28  Science & Technology
16  30                Movies
17  31       Anime/Animation
18  32      Action/Adventure
19  33              Classics
20  34                Comedy
21  35           Documentary
22  36                 Drama
23  37                Family
24  38               Foreign
25  39                Horror
26  40        Sci-Fi/Fantasy
27  41              Thriller
28  42                Shorts
29  43                 Shows
30  44              Trailers

In [16]:
dfCAcategory1.shape

(31, 2)

Berdasarkan pengecekan data kategori region Canada memiliki dimensi <b>31 baris x 2 kolom</b>.

### Data Kategori Video Region Great Britain
Berikut adalah data kategori video region Great Britain setelah diubah menjadi dataframe :

In [17]:
dfGBcategory1

id                 title
0    1      Film & Animation
1    2      Autos & Vehicles
2   10                 Music
3   15        Pets & Animals
4   17                Sports
5   18          Short Movies
6   19       Travel & Events
7   20                Gaming
8   21         Videoblogging
9   22        People & Blogs
10  23                Comedy
11  24         Entertainment
12  25       News & Politics
13  26         Howto & Style
14  27             Education
15  28  Science & Technology
16  30                Movies
17  31       Anime/Animation
18  32      Action/Adventure
19  33              Classics
20  34                Comedy
21  35           Documentary
22  36                 Drama
23  37                Family
24  38               Foreign
25  39                Horror
26  40        Sci-Fi/Fantasy
27  41              Thriller
28  42                Shorts
29  43                 Shows
30  44              Trailers

In [18]:
dfGBcategory1.shape

(31, 2)

Berdasarkan pengecekan data kategori region Great Britain memiliki dimensi <b>31 baris x 2 kolom</b>.

### Data Kategori Video Region United States
Berikut adalah data kategori video region Great Britain setelah diubah menjadi dataframe :

In [19]:
dfUScategory1

id                  title
0    1       Film & Animation
1    2       Autos & Vehicles
2   10                  Music
3   15         Pets & Animals
4   17                 Sports
5   18           Short Movies
6   19        Travel & Events
7   20                 Gaming
8   21          Videoblogging
9   22         People & Blogs
10  23                 Comedy
11  24          Entertainment
12  25        News & Politics
13  26          Howto & Style
14  27              Education
15  28   Science & Technology
16  29  Nonprofits & Activism
17  30                 Movies
18  31        Anime/Animation
19  32       Action/Adventure
20  33               Classics
21  34                 Comedy
22  35            Documentary
23  36                  Drama
24  37                 Family
25  38                Foreign
26  39                 Horror
27  40         Sci-Fi/Fantasy
28  41               Thriller
29  42                 Shorts
30  43                  Shows
31  44               Trailers

In [20]:
dfUScategory1.shape

(32, 2)

Berdasarkan pengecekan data kategori region Canada memiliki dimensi <b>31 baris x 2 kolom</b>.

### Persamaan & Perbedaan Data Kategori dari 3 region
Berdasarkan pengecekan dataframe kategori region Canada dan Great Britain adalah identik, sedangkan dataframe kategori dari United States berbeda dari keduanya.

In [21]:
dfCAcategory1.equals(dfGBcategory1)

True

In [22]:
dfGBcategory1.equals(dfUScategory1)

False

In [23]:
dfCek = pd.concat([dfCAcategory1,dfGBcategory1,dfUScategory1]).drop_duplicates(keep=False)
dfCek

id                  title
16  29  Nonprofits & Activism

Perbedaan yang didapatkan dengan melakukan concat adalah dataframe kategori United States memiliki satu title tambahan yaitu "Nonprofits & Activism" pada index ke-16.

Maka dari itu diputuskan bahwa data kategori dari 3 region ini <b>di-merge menjadi 1 dataframe</b> saja dengan pd.merge(x,y) dengan keterangan sebagai berikut :
- x = dataframe kategori US
- y = dataframe kategori CA

### Merge Data Kategori

In [24]:
#merge
dfCategory = pd.merge(dfCAcategory1,dfUScategory1, how="right")

dfCategory

id                  title
0    1       Film & Animation
1    2       Autos & Vehicles
2   10                  Music
3   15         Pets & Animals
4   17                 Sports
5   18           Short Movies
6   19        Travel & Events
7   20                 Gaming
8   21          Videoblogging
9   22         People & Blogs
10  23                 Comedy
11  24          Entertainment
12  25        News & Politics
13  26          Howto & Style
14  27              Education
15  28   Science & Technology
16  30                 Movies
17  31        Anime/Animation
18  32       Action/Adventure
19  33               Classics
20  34                 Comedy
21  35            Documentary
22  36                  Drama
23  37                 Family
24  38                Foreign
25  39                 Horror
26  40         Sci-Fi/Fantasy
27  41               Thriller
28  42                 Shorts
29  43                  Shows
30  44               Trailers
31  29  Nonprofits & Activism

Dataframe 3 region disatukan menjadi dataframe <b>dfCategory</b>.

### Sortir Data Kategori

Setelah dilakukan merge, didapatkan bahwa "id" tidak berurutan. Maka dari itu dilakukan sorting/sortir dengan .sort_values(). Tetapi agar dapat disortir, kolom "id" harus di-state terlebih dahulu bahwa data di kolom tersebut bertipe integer. Setelah, "id" disortir, perlu dilakukan pengindeksan ulang agar index menjadi berurutan kembali.

Berikut adalah <b>Data Kategori Utama</b> :

In [25]:
#Menstate kolom id sebagai data bertipe integer agar dapat disortir
dfCategory['id'] = dfCategory['id'].astype('int')

#Sortir
dfCategory.sort_values("id", ascending=True)

#Pengindexan ulang
dfCategory = dfCategory.reset_index(drop=True)

print("")
print("DATA KATEGORI UTAMA")
dfCategory


DATA KATEGORI UTAMA


id                  title
0    1       Film & Animation
1    2       Autos & Vehicles
2   10                  Music
3   15         Pets & Animals
4   17                 Sports
5   18           Short Movies
6   19        Travel & Events
7   20                 Gaming
8   21          Videoblogging
9   22         People & Blogs
10  23                 Comedy
11  24          Entertainment
12  25        News & Politics
13  26          Howto & Style
14  27              Education
15  28   Science & Technology
16  30                 Movies
17  31        Anime/Animation
18  32       Action/Adventure
19  33               Classics
20  34                 Comedy
21  35            Documentary
22  36                  Drama
23  37                 Family
24  38                Foreign
25  39                 Horror
26  40         Sci-Fi/Fantasy
27  41               Thriller
28  42                 Shorts
29  43                  Shows
30  44               Trailers
31  29  Nonprofits & Activism

In [26]:
dfCategory.shape

(32, 2)

Berdasarkan pengecekan data kategori memiliki dimensi <b>32 baris x 2 kolom</b>

## Statistik Dari Setiap Kategori di Data Video
Kita perlu mengetahui statistik berikut untuk setiap kategori :
#### 1. Data tertinggi, terendah, dan rata-rata viewer
#### 2. Data tertinggi, terendah, dan rata-rata likes
#### 3. Data tertinggi, terendah, dan rata-rata dislikes
#### 5. Data tertinggi, terendah, dan rata-rata jumlah komentar

### Pertama kita perlu mengetahui kategori apa yang ada pada masing-masing region berdasarkan data video

In [27]:
# drop kolom yang tidak perlu
dfCA = dfCAvideos.drop(['trending_date','channel_title','publish_time','tags','thumbnail_link','comments_disabled','ratings_disabled','video_error_or_removed','description'], axis=1)
dfGB = dfGBvideos.drop(['trending_date','channel_title','publish_time','tags','thumbnail_link','comments_disabled','ratings_disabled','video_error_or_removed','description'], axis=1)
dfUS = dfUSvideos.drop(['trending_date','channel_title','publish_time','tags','thumbnail_link','comments_disabled','ratings_disabled','video_error_or_removed','description'], axis=1)

In [28]:
# mengelompokan berdasarkan id kategori
gCAvideo = dfCA.groupby('category_id')
gGBvideo = dfGB.groupby('category_id')
gUSvideo = dfUS.groupby('category_id')

#### Menemukan value unique pada id kategori region Canada

In [29]:
# menemukan value yang unik pada id kategori region canada
dfCA.category_id.unique()

a = dfCA.category_id.unique()
a.sort(axis=0)
a

array([ 1,  2, 10, 15, 17, 19, 20, 22, 23, 24, 25, 26, 27, 28, 29, 30, 43],
      dtype=int64)

#### Menemukan value unique pada id kategori region Great Britain

In [30]:
# menemukan value yang unik pada id kategori region great britain
dfGB.category_id.unique()

b = dfGB.category_id.unique()
b.sort(axis=0)
b

array([ 1,  2, 10, 15, 17, 19, 20, 22, 23, 24, 25, 26, 27, 28, 29, 43],
      dtype=int64)

#### Menemukan value unique pada id kategori region United States

In [31]:
# menemukan value yang unik pada id kategori region us
dfUS.category_id.unique()

c = dfUS.category_id.unique()
c.sort(axis=0)
c

array([ 1,  2, 10, 15, 17, 19, 20, 22, 23, 24, 25, 26, 27, 28, 29, 43],
      dtype=int64)

In [32]:
# membuat data list dari array
data_cat_ca = {'id': dfCA.category_id.unique()}
data_cat_gb = {'id': dfGB.category_id.unique()} 
data_cat_us = {'id': dfUS.category_id.unique()} 

# memasukan data kategori ke dataframe
dfCAcat = pd.DataFrame(data_cat_ca)
dfGBcat = pd.DataFrame(data_cat_gb)
dfUScat = pd.DataFrame(data_cat_us)

In [33]:
# melakukan merge data kategori canada dengan dataframe kategori utama
dfCAcat1 = pd.merge(dfCategory,dfCAcat)
print("")
print("DATA KATEGORI DI CANADA")
dfCAcat1


DATA KATEGORI DI CANADA


id                  title
0    1       Film & Animation
1    2       Autos & Vehicles
2   10                  Music
3   15         Pets & Animals
4   17                 Sports
5   19        Travel & Events
6   20                 Gaming
7   22         People & Blogs
8   23                 Comedy
9   24          Entertainment
10  25        News & Politics
11  26          Howto & Style
12  27              Education
13  28   Science & Technology
14  30                 Movies
15  43                  Shows
16  29  Nonprofits & Activism

In [34]:
# melakukan merge data kategori great britain dengan dataframe kategori utama
dfGBcat1 = pd.merge(dfCategory,dfGBcat)
print("")
print("DATA KATEGORI DI GREAT BRITAIN")
dfGBcat1


DATA KATEGORI DI GREAT BRITAIN


id                  title
0    1       Film & Animation
1    2       Autos & Vehicles
2   10                  Music
3   15         Pets & Animals
4   17                 Sports
5   19        Travel & Events
6   20                 Gaming
7   22         People & Blogs
8   23                 Comedy
9   24          Entertainment
10  25        News & Politics
11  26          Howto & Style
12  27              Education
13  28   Science & Technology
14  43                  Shows
15  29  Nonprofits & Activism

In [35]:
# melakukan merge data kategori united states dengan dataframe kategori utama
dfUScat1 = pd.merge(dfCategory,dfUScat)
print("")
print("DATA KATEGORI DI UNITED STATES")
dfUScat1


DATA KATEGORI DI UNITED STATES


id                  title
0    1       Film & Animation
1    2       Autos & Vehicles
2   10                  Music
3   15         Pets & Animals
4   17                 Sports
5   19        Travel & Events
6   20                 Gaming
7   22         People & Blogs
8   23                 Comedy
9   24          Entertainment
10  25        News & Politics
11  26          Howto & Style
12  27              Education
13  28   Science & Technology
14  43                  Shows
15  29  Nonprofits & Activism

In [36]:
dfGBcat1.equals(dfUScat1)

True

In [37]:
dfGBcat1.equals(dfCAcat1)

False

### Ditemukan bahwa kategori video yang ada di data video Great Britain dan United States adalah identik, yaitu :
- 1	(Film & Animation)
- 2	(Autos & Vehicles)
- 10 (Music)
- 15 (Pets & Animals)
- 17 (Sports)
- 19 (Travel & Events)
- 20 (Gaming)
- 22 (People & Blogs)
- 23 (Comedy)
- 24 (Entertainment)
- 25 (News & Politics)
- 26 (Howto & Style)
- 27 (Education)
- 28 (Science & Technology)
- 43 (Shows)
- 29 (Nonprofits & Activism)

### Sedangkan data video Canada memiliki 1 kategori yang berbeda dari Great Britain dan United States, yaitu :
- 30 (Movies)

## Statistik Data Video Region Canada
Di sini kita akan menemukan statistik data video dari region Canada menggunakan list <b>gCAvideo</b>.<br/>
Untuk menemukan rata-rata digunakan fungsi <b>.max()</b> pada data grup id kategori yang diinginkan. Untuk menemukan value tertinggi digunakan <b>.max()</b> dan untuk menemukan value terkecil digunakan <b>.min()</b>

#### 1A. Data Viewer Canada

In [38]:
# memasukkan data ke dalam list
data_view_ca = {
    
    # a adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Canada
    'id' : a,
    'highest_views':[gCAvideo.get_group(1)['views'].max(),
                     gCAvideo.get_group(2)['views'].max(),
                     gCAvideo.get_group(10)['views'].max(),
                     gCAvideo.get_group(15)['views'].max(),
                     gCAvideo.get_group(17)['views'].max(),
                     gCAvideo.get_group(19)['views'].max(),
                     gCAvideo.get_group(20)['views'].max(),
                     gCAvideo.get_group(22)['views'].max(),
                     gCAvideo.get_group(23)['views'].max(),
                     gCAvideo.get_group(24)['views'].max(),
                     gCAvideo.get_group(25)['views'].max(),
                     gCAvideo.get_group(26)['views'].max(),
                     gCAvideo.get_group(27)['views'].max(),
                     gCAvideo.get_group(28)['views'].max(),
                     gCAvideo.get_group(30)['views'].max(),
                     gCAvideo.get_group(43)['views'].max(),
                     gCAvideo.get_group(29)['views'].max()],
    
    'lowest_views':[gCAvideo.get_group(1)['views'].min(),
                     gCAvideo.get_group(2)['views'].min(),
                     gCAvideo.get_group(10)['views'].min(),
                     gCAvideo.get_group(15)['views'].min(),
                     gCAvideo.get_group(17)['views'].min(),
                     gCAvideo.get_group(19)['views'].min(),
                     gCAvideo.get_group(20)['views'].min(),
                     gCAvideo.get_group(22)['views'].min(),
                     gCAvideo.get_group(23)['views'].min(),
                     gCAvideo.get_group(24)['views'].min(),
                     gCAvideo.get_group(25)['views'].min(),
                     gCAvideo.get_group(26)['views'].min(),
                     gCAvideo.get_group(27)['views'].min(),
                     gCAvideo.get_group(28)['views'].min(),
                     gCAvideo.get_group(30)['views'].min(),
                     gCAvideo.get_group(43)['views'].min(),
                     gCAvideo.get_group(29)['views'].min()],
    
    'average_views':[gCAvideo.get_group(1)['views'].mean(),
                     gCAvideo.get_group(2)['views'].mean(),
                     gCAvideo.get_group(10)['views'].mean(),
                     gCAvideo.get_group(15)['views'].mean(),
                     gCAvideo.get_group(17)['views'].mean(),
                     gCAvideo.get_group(19)['views'].mean(),
                     gCAvideo.get_group(20)['views'].mean(),
                     gCAvideo.get_group(22)['views'].mean(),
                     gCAvideo.get_group(23)['views'].mean(),
                     gCAvideo.get_group(24)['views'].mean(),
                     gCAvideo.get_group(25)['views'].mean(),
                     gCAvideo.get_group(26)['views'].mean(),
                     gCAvideo.get_group(27)['views'].mean(),
                     gCAvideo.get_group(28)['views'].mean(),
                     gCAvideo.get_group(30)['views'].mean(),
                     gCAvideo.get_group(43)['views'].mean(),
                     gCAvideo.get_group(29)['views'].mean()
                   ]}

dfViewCA = pd.DataFrame(data_view_ca)

In [39]:
dfViewCA = pd.merge(dfCAcat1,dfViewCA,on="id")

print("")
print("DATA VIEWER SETIAP KATEGORI DI REGION CANADA")
dfViewCA


DATA VIEWER SETIAP KATEGORI DI REGION CANADA


id                  title  highest_views  lowest_views  average_views
0    1       Film & Animation       27696924          2274 1426728.565049
1    2       Autos & Vehicles       11004698          3059  566759.416431
2   10                  Music       98938809          3201 3532524.844278
3   15         Pets & Animals        5954189          8171  638461.173442
4   17                 Sports       24117244          1023 1075583.849300
5   19        Travel & Events        4169455          3144  366701.408163
6   20                 Gaming       14589314          3494  923759.215030
7   22         People & Blogs       51243149          1187  786413.623873
8   23                 Comedy       14228882          1295  982888.625762
9   24          Entertainment      137843120           733 1016371.683072
10  25        News & Politics       11510500          1000  388220.736475
11  26          Howto & Style       35696481          3564  782683.911809
12  27              Education       12100921          1141  536602.767911
13  28   Science & Technology       36778884          3933 1233844.520346
14  30                 Movies        1386464         42240  577012.161290
15  43                  Shows       22387656          1898 1562184.094595
16  29  Nonprofits & Activism        5661965        225528 2853415.000000

#### 2A. Data Likes Video di Canada

In [40]:
# memasukkan data ke dalam list
data_likes_ca = {
    
    # a adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Canada
    'id' : a,
    'highest_likes':[gCAvideo.get_group(1)['likes'].max(),
                     gCAvideo.get_group(2)['likes'].max(),
                     gCAvideo.get_group(10)['likes'].max(),
                     gCAvideo.get_group(15)['likes'].max(),
                     gCAvideo.get_group(17)['likes'].max(),
                     gCAvideo.get_group(19)['likes'].max(),
                     gCAvideo.get_group(20)['likes'].max(),
                     gCAvideo.get_group(22)['likes'].max(),
                     gCAvideo.get_group(23)['likes'].max(),
                     gCAvideo.get_group(24)['likes'].max(),
                     gCAvideo.get_group(25)['likes'].max(),
                     gCAvideo.get_group(26)['likes'].max(),
                     gCAvideo.get_group(27)['likes'].max(),
                     gCAvideo.get_group(28)['likes'].max(),
                     gCAvideo.get_group(30)['likes'].max(),
                     gCAvideo.get_group(43)['likes'].max(),
                     gCAvideo.get_group(29)['likes'].max()],
    
    'lowest_likes':[gCAvideo.get_group(1)['likes'].min(),
                     gCAvideo.get_group(2)['likes'].min(),
                     gCAvideo.get_group(10)['likes'].min(),
                     gCAvideo.get_group(15)['likes'].min(),
                     gCAvideo.get_group(17)['likes'].min(),
                     gCAvideo.get_group(19)['likes'].min(),
                     gCAvideo.get_group(20)['likes'].min(),
                     gCAvideo.get_group(22)['likes'].min(),
                     gCAvideo.get_group(23)['likes'].min(),
                     gCAvideo.get_group(24)['likes'].min(),
                     gCAvideo.get_group(25)['likes'].min(),
                     gCAvideo.get_group(26)['likes'].min(),
                     gCAvideo.get_group(27)['likes'].min(),
                     gCAvideo.get_group(28)['likes'].min(),
                     gCAvideo.get_group(30)['likes'].min(),
                     gCAvideo.get_group(43)['likes'].min(),
                     gCAvideo.get_group(29)['likes'].min()],
    
    'average_likes':[gCAvideo.get_group(1)['likes'].mean(),
                     gCAvideo.get_group(2)['likes'].mean(),
                     gCAvideo.get_group(10)['likes'].mean(),
                     gCAvideo.get_group(15)['likes'].mean(),
                     gCAvideo.get_group(17)['likes'].mean(),
                     gCAvideo.get_group(19)['likes'].mean(),
                     gCAvideo.get_group(20)['likes'].mean(),
                     gCAvideo.get_group(22)['likes'].mean(),
                     gCAvideo.get_group(23)['likes'].mean(),
                     gCAvideo.get_group(24)['likes'].mean(),
                     gCAvideo.get_group(25)['likes'].mean(),
                     gCAvideo.get_group(26)['likes'].mean(),
                     gCAvideo.get_group(27)['likes'].mean(),
                     gCAvideo.get_group(28)['likes'].mean(),
                     gCAvideo.get_group(30)['likes'].mean(),
                     gCAvideo.get_group(43)['likes'].mean(),
                     gCAvideo.get_group(29)['likes'].mean()
                   ]}

dfLikesCA = pd.DataFrame(data_likes_ca)

In [41]:
dfLikesCA = pd.merge(dfCAcat1,dfLikesCA,on="id")

print("")
print("DATA LIKES SETIAP KATEGORI DI REGION CANADA")
dfLikesCA


DATA LIKES SETIAP KATEGORI DI REGION CANADA


id                  title  highest_likes  lowest_likes  average_likes
0    1       Film & Animation         532660             0   37767.962621
1    2       Autos & Vehicles          86125            40   13405.422096
2   10                  Music        5053338             0  151285.856339
3   15         Pets & Animals         193092           215   23931.533875
4   17                 Sports         926660             0   21330.972013
5   19        Travel & Events         135057             4    9568.658163
6   20                 Gaming         340240             0   38487.017857
7   22         People & Blogs        1967910             0   25673.219001
8   23                 Comedy         831516             4   51960.422475
9   24          Entertainment        3014479             0   28789.341536
10  25        News & Politics         552150             0    7863.682135
11  26          Howto & Style         443219             0   25502.293971
12  27              Education         282663             0   21162.867810
13  28   Science & Technology         521873             0   37068.916017
14  30                 Movies          25196           122    4752.120968
15  43                  Shows        1919980             0  138413.689189
16  29  Nonprofits & Activism          67232          9336   42150.833333

#### 3A. Data Dislikes Video di Canada

In [42]:
# memasukkan data ke dalam list
data_dislikes_ca = {
    
    # a adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Canada
    'id' : a,
    'highest_dislikes':[gCAvideo.get_group(1)['dislikes'].max(),
                     gCAvideo.get_group(2)['dislikes'].max(),
                     gCAvideo.get_group(10)['dislikes'].max(),
                     gCAvideo.get_group(15)['dislikes'].max(),
                     gCAvideo.get_group(17)['dislikes'].max(),
                     gCAvideo.get_group(19)['dislikes'].max(),
                     gCAvideo.get_group(20)['dislikes'].max(),
                     gCAvideo.get_group(22)['dislikes'].max(),
                     gCAvideo.get_group(23)['dislikes'].max(),
                     gCAvideo.get_group(24)['dislikes'].max(),
                     gCAvideo.get_group(25)['dislikes'].max(),
                     gCAvideo.get_group(26)['dislikes'].max(),
                     gCAvideo.get_group(27)['dislikes'].max(),
                     gCAvideo.get_group(28)['dislikes'].max(),
                     gCAvideo.get_group(30)['dislikes'].max(),
                     gCAvideo.get_group(43)['dislikes'].max(),
                     gCAvideo.get_group(29)['dislikes'].max()],
    
    'lowest_dislikes':[gCAvideo.get_group(1)['dislikes'].min(),
                     gCAvideo.get_group(2)['dislikes'].min(),
                     gCAvideo.get_group(10)['dislikes'].min(),
                     gCAvideo.get_group(15)['dislikes'].min(),
                     gCAvideo.get_group(17)['dislikes'].min(),
                     gCAvideo.get_group(19)['dislikes'].min(),
                     gCAvideo.get_group(20)['dislikes'].min(),
                     gCAvideo.get_group(22)['dislikes'].min(),
                     gCAvideo.get_group(23)['dislikes'].min(),
                     gCAvideo.get_group(24)['dislikes'].min(),
                     gCAvideo.get_group(25)['dislikes'].min(),
                     gCAvideo.get_group(26)['dislikes'].min(),
                     gCAvideo.get_group(27)['dislikes'].min(),
                     gCAvideo.get_group(28)['dislikes'].min(),
                     gCAvideo.get_group(30)['dislikes'].min(),
                     gCAvideo.get_group(43)['dislikes'].min(),
                     gCAvideo.get_group(29)['dislikes'].min()],
    
    'average_dislikes':[gCAvideo.get_group(1)['dislikes'].mean(),
                     gCAvideo.get_group(2)['dislikes'].mean(),
                     gCAvideo.get_group(10)['dislikes'].mean(),
                     gCAvideo.get_group(15)['dislikes'].mean(),
                     gCAvideo.get_group(17)['dislikes'].mean(),
                     gCAvideo.get_group(19)['dislikes'].mean(),
                     gCAvideo.get_group(20)['dislikes'].mean(),
                     gCAvideo.get_group(22)['dislikes'].mean(),
                     gCAvideo.get_group(23)['dislikes'].mean(),
                     gCAvideo.get_group(24)['dislikes'].mean(),
                     gCAvideo.get_group(25)['dislikes'].mean(),
                     gCAvideo.get_group(26)['dislikes'].mean(),
                     gCAvideo.get_group(27)['dislikes'].mean(),
                     gCAvideo.get_group(28)['dislikes'].mean(),
                     gCAvideo.get_group(30)['dislikes'].mean(),
                     gCAvideo.get_group(43)['dislikes'].mean(),
                     gCAvideo.get_group(29)['dislikes'].mean()
                   ]}

dfDislikesCA = pd.DataFrame(data_dislikes_ca)

In [43]:
dfDislikesCA = pd.merge(dfCAcat1,dfDislikesCA,on="id")

print("")
print("DATA DISLIKES SETIAP KATEGORI DI REGION CANADA")
dfDislikesCA


DATA DISLIKES SETIAP KATEGORI DI REGION CANADA


id                  title  highest_dislikes  lowest_dislikes  \
0    1       Film & Animation             60025                0   
1    2       Autos & Vehicles             12290                1   
2   10                  Music            165854                0   
3   15         Pets & Animals              7732                5   
4   17                 Sports            164693                0   
5   19        Travel & Events              7688                0   
6   20                 Gaming            174645                0   
7   22         People & Blogs            504340                0   
8   23                 Comedy             43698                0   
9   24          Entertainment           1602383                0   
10  25        News & Politics            126649                0   
11  26          Howto & Style             48163                0   
12  27              Education             13561                0   
13  28   Science & Technology             45910                0   
14  30                 Movies              2864               38   
15  43                  Shows            461660                0   
16  29  Nonprofits & Activism              3784              222   

    average_dislikes  
0        1510.319903  
1         560.271955  
2        5922.859823  
3         501.363144  
4        1179.177969  
5         331.278061  
6        1820.596726  
7        2309.572716  
8        1423.937185  
9        1917.447253  
10        784.813898  
11        862.242152  
12        800.653885  
13       1550.781818  
14        971.258065  
15      31429.824324  
16       2046.666667

#### 4A. Data Jumlah Komentar Video di Canada

In [44]:
# memasukkan data ke dalam list
data_comments_ca = {
    
    # a adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Canada
    'id' : a,
    'highest_comments':[gCAvideo.get_group(1)['comment_count'].max(),
                     gCAvideo.get_group(2)['comment_count'].max(),
                     gCAvideo.get_group(10)['comment_count'].max(),
                     gCAvideo.get_group(15)['comment_count'].max(),
                     gCAvideo.get_group(17)['comment_count'].max(),
                     gCAvideo.get_group(19)['comment_count'].max(),
                     gCAvideo.get_group(20)['comment_count'].max(),
                     gCAvideo.get_group(22)['comment_count'].max(),
                     gCAvideo.get_group(23)['comment_count'].max(),
                     gCAvideo.get_group(24)['comment_count'].max(),
                     gCAvideo.get_group(25)['comment_count'].max(),
                     gCAvideo.get_group(26)['comment_count'].max(),
                     gCAvideo.get_group(27)['comment_count'].max(),
                     gCAvideo.get_group(28)['comment_count'].max(),
                     gCAvideo.get_group(30)['comment_count'].max(),
                     gCAvideo.get_group(43)['comment_count'].max(),
                     gCAvideo.get_group(29)['comment_count'].max()],
    
    'lowest_comments':[gCAvideo.get_group(1)['comment_count'].min(),
                     gCAvideo.get_group(2)['comment_count'].min(),
                     gCAvideo.get_group(10)['comment_count'].min(),
                     gCAvideo.get_group(15)['comment_count'].min(),
                     gCAvideo.get_group(17)['comment_count'].min(),
                     gCAvideo.get_group(19)['comment_count'].min(),
                     gCAvideo.get_group(20)['comment_count'].min(),
                     gCAvideo.get_group(22)['comment_count'].min(),
                     gCAvideo.get_group(23)['comment_count'].min(),
                     gCAvideo.get_group(24)['comment_count'].min(),
                     gCAvideo.get_group(25)['comment_count'].min(),
                     gCAvideo.get_group(26)['comment_count'].min(),
                     gCAvideo.get_group(27)['comment_count'].min(),
                     gCAvideo.get_group(28)['comment_count'].min(),
                     gCAvideo.get_group(30)['comment_count'].min(),
                     gCAvideo.get_group(43)['comment_count'].min(),
                     gCAvideo.get_group(29)['comment_count'].min()],
    
    'average_comments':[gCAvideo.get_group(1)['comment_count'].mean(),
                     gCAvideo.get_group(2)['comment_count'].mean(),
                     gCAvideo.get_group(10)['comment_count'].mean(),
                     gCAvideo.get_group(15)['comment_count'].mean(),
                     gCAvideo.get_group(17)['comment_count'].mean(),
                     gCAvideo.get_group(19)['comment_count'].mean(),
                     gCAvideo.get_group(20)['comment_count'].mean(),
                     gCAvideo.get_group(22)['comment_count'].mean(),
                     gCAvideo.get_group(23)['comment_count'].mean(),
                     gCAvideo.get_group(24)['comment_count'].mean(),
                     gCAvideo.get_group(25)['comment_count'].mean(),
                     gCAvideo.get_group(26)['comment_count'].mean(),
                     gCAvideo.get_group(27)['comment_count'].mean(),
                     gCAvideo.get_group(28)['comment_count'].mean(),
                     gCAvideo.get_group(30)['comment_count'].mean(),
                     gCAvideo.get_group(43)['comment_count'].mean(),
                     gCAvideo.get_group(29)['comment_count'].mean()
                   ]}

dfCommentsCA = pd.DataFrame(data_comments_ca)

In [45]:
dfCommentsCA = pd.merge(dfCAcat1,dfCommentsCA,on="id")

print("")
print("DATA COMMENTS SETIAP KATEGORI DI REGION CANADA")
dfCommentsCA


DATA COMMENTS SETIAP KATEGORI DI REGION CANADA


id                  title  highest_comments  lowest_comments  \
0    1       Film & Animation            105587                0   
1    2       Autos & Vehicles             29574                0   
2   10                  Music           1114800                0   
3   15         Pets & Animals             45306                0   
4   17                 Sports             63560                0   
5   19        Travel & Events             13612                0   
6   20                 Gaming            122659                0   
7   22         People & Blogs            349112                0   
8   23                 Comedy             83116                0   
9   24          Entertainment            827755                0   
10  25        News & Politics             64291                0   
11  26          Howto & Style            285909                0   
12  27              Education             88591                0   
13  28   Science & Technology            162654                0   
14  30                 Movies              3628               27   
15  43                  Shows            625010                0   
16  29  Nonprofits & Activism              3851              438   

    average_comments  
0        4896.394660  
1        2878.719547  
2       14053.940499  
3        4941.837398  
4        2748.956584  
5        1556.002551  
6        7191.136161  
7        4097.003898  
8        5205.082428  
9        4112.187644  
10       2226.445780  
11       4156.909317  
12       2995.374369  
13       6241.748052  
14        622.927419  
15      43547.851351  
16       2329.833333

## Statistik Data Video Region Great Britain
Di sini kita akan menemukan statistik data video dari region Great Britain menggunakan list <b>gGBvideo</b>.<br/>
Untuk menemukan rata-rata digunakan fungsi <b>.max()</b> pada data grup id kategori yang diinginkan. Untuk menemukan value tertinggi digunakan <b>.max()</b> dan untuk menemukan value terkecil digunakan <b>.min()</b>

Perbedaan data video di Great Britain adalah tidak ada kategori 30 / Movies seperti pada data Canada.

#### 1B. Data Viewer Great Britain

In [46]:
# memasukkan data ke dalam list
data_view_gb = {
    
    # b adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Great Britain
    'id' : b,
    'highest_views':[gGBvideo.get_group(1)['views'].max(),
                     gGBvideo.get_group(2)['views'].max(),
                     gGBvideo.get_group(10)['views'].max(),
                     gGBvideo.get_group(15)['views'].max(),
                     gGBvideo.get_group(17)['views'].max(),
                     gGBvideo.get_group(19)['views'].max(),
                     gGBvideo.get_group(20)['views'].max(),
                     gGBvideo.get_group(22)['views'].max(),
                     gGBvideo.get_group(23)['views'].max(),
                     gGBvideo.get_group(24)['views'].max(),
                     gGBvideo.get_group(25)['views'].max(),
                     gGBvideo.get_group(26)['views'].max(),
                     gGBvideo.get_group(27)['views'].max(),
                     gGBvideo.get_group(28)['views'].max(),
                     gGBvideo.get_group(43)['views'].max(),
                     gGBvideo.get_group(29)['views'].max()],
    
    'lowest_views':[gGBvideo.get_group(1)['views'].min(),
                     gGBvideo.get_group(2)['views'].min(),
                     gGBvideo.get_group(10)['views'].min(),
                     gGBvideo.get_group(15)['views'].min(),
                     gGBvideo.get_group(17)['views'].min(),
                     gGBvideo.get_group(19)['views'].min(),
                     gGBvideo.get_group(20)['views'].min(),
                     gGBvideo.get_group(22)['views'].min(),
                     gGBvideo.get_group(23)['views'].min(),
                     gGBvideo.get_group(24)['views'].min(),
                     gGBvideo.get_group(25)['views'].min(),
                     gGBvideo.get_group(26)['views'].min(),
                     gGBvideo.get_group(27)['views'].min(),
                     gGBvideo.get_group(28)['views'].min(),
                     gGBvideo.get_group(43)['views'].min(),
                     gGBvideo.get_group(29)['views'].min()],
    
    'average_views':[gGBvideo.get_group(1)['views'].mean(),
                     gGBvideo.get_group(2)['views'].mean(),
                     gGBvideo.get_group(10)['views'].mean(),
                     gGBvideo.get_group(15)['views'].mean(),
                     gGBvideo.get_group(17)['views'].mean(),
                     gGBvideo.get_group(19)['views'].mean(),
                     gGBvideo.get_group(20)['views'].mean(),
                     gGBvideo.get_group(22)['views'].mean(),
                     gGBvideo.get_group(23)['views'].mean(),
                     gGBvideo.get_group(24)['views'].mean(),
                     gGBvideo.get_group(25)['views'].mean(),
                     gGBvideo.get_group(26)['views'].mean(),
                     gGBvideo.get_group(27)['views'].mean(),
                     gGBvideo.get_group(28)['views'].mean(),
                     gGBvideo.get_group(43)['views'].mean(),
                     gGBvideo.get_group(29)['views'].mean()
                   ]}

dfViewGB = pd.DataFrame(data_view_gb)

In [47]:
dfViewCA = pd.merge(dfCAcat1,dfViewCA,on="id")

print("")
print("DATA VIEWER SETIAP KATEGORI DI REGION CANADA")
dfViewCA


DATA VIEWER SETIAP KATEGORI DI REGION CANADA


id                title_x                title_y  highest_views  \
0    1       Film & Animation       Film & Animation       27696924   
1    2       Autos & Vehicles       Autos & Vehicles       11004698   
2   10                  Music                  Music       98938809   
3   15         Pets & Animals         Pets & Animals        5954189   
4   17                 Sports                 Sports       24117244   
5   19        Travel & Events        Travel & Events        4169455   
6   20                 Gaming                 Gaming       14589314   
7   22         People & Blogs         People & Blogs       51243149   
8   23                 Comedy                 Comedy       14228882   
9   24          Entertainment          Entertainment      137843120   
10  25        News & Politics        News & Politics       11510500   
11  26          Howto & Style          Howto & Style       35696481   
12  27              Education              Education       12100921   
13  28   Science & Technology   Science & Technology       36778884   
14  30                 Movies                 Movies        1386464   
15  43                  Shows                  Shows       22387656   
16  29  Nonprofits & Activism  Nonprofits & Activism        5661965   

    lowest_views  average_views  
0           2274 1426728.565049  
1           3059  566759.416431  
2           3201 3532524.844278  
3           8171  638461.173442  
4           1023 1075583.849300  
5           3144  366701.408163  
6           3494  923759.215030  
7           1187  786413.623873  
8           1295  982888.625762  
9            733 1016371.683072  
10          1000  388220.736475  
11          3564  782683.911809  
12          1141  536602.767911  
13          3933 1233844.520346  
14         42240  577012.161290  
15          1898 1562184.094595  
16        225528 2853415.000000

In [48]:
dfViewGB = pd.merge(dfGBcat1,dfViewGB,on="id")

print("")
print("DATA VIEWER SETIAP KATEGORI DI REGION GREAT BRITAIN")
dfViewGB


DATA VIEWER SETIAP KATEGORI DI REGION GREAT BRITAIN


id                  title  highest_views  lowest_views   average_views
0    1       Film & Animation       42560014          2441  3245132.955374
1    2       Autos & Vehicles        5362914          1858  1487392.458333
2   10                  Music      424538912          2152 12444442.690781
3   15         Pets & Animals        6416920         25232  1013708.938202
4   17                 Sports       22975455          1930  1752352.487677
5   19        Travel & Events        6362355          3254  1383177.010417
6   20                 Gaming       18158133          7350   976856.561521
7   22         People & Blogs       62338362          1184  1966605.546480
8   23                 Comedy       43460605         10676  2167430.056346
9   24          Entertainment      169884583          2650  3264607.961530
10  25        News & Politics       18994966           851  1150311.326531
11  26          Howto & Style        8721841          1505   617300.428942
12  27              Education       11240174          8851   972189.402626
13  28   Science & Technology       21530325         38353  3168703.196911
14  43                  Shows       26703269         19270  3919980.688889
15  29  Nonprofits & Activism        1709880        876996  1429164.150000

#### 2B. Data Likes Video di Great Britain

In [49]:
# memasukkan data ke dalam list
data_likes_gb = {
    
    # b adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Great Britain
    'id' : b,
    'highest_likes':[gGBvideo.get_group(1)['likes'].max(),
                     gGBvideo.get_group(2)['likes'].max(),
                     gGBvideo.get_group(10)['likes'].max(),
                     gGBvideo.get_group(15)['likes'].max(),
                     gGBvideo.get_group(17)['likes'].max(),
                     gGBvideo.get_group(19)['likes'].max(),
                     gGBvideo.get_group(20)['likes'].max(),
                     gGBvideo.get_group(22)['likes'].max(),
                     gGBvideo.get_group(23)['likes'].max(),
                     gGBvideo.get_group(24)['likes'].max(),
                     gGBvideo.get_group(25)['likes'].max(),
                     gGBvideo.get_group(26)['likes'].max(),
                     gGBvideo.get_group(27)['likes'].max(),
                     gGBvideo.get_group(28)['likes'].max(),
                     gGBvideo.get_group(43)['likes'].max(),
                     gGBvideo.get_group(29)['likes'].max()],
    
    'lowest_likes':[gGBvideo.get_group(1)['likes'].min(),
                     gGBvideo.get_group(2)['likes'].min(),
                     gGBvideo.get_group(10)['likes'].min(),
                     gGBvideo.get_group(15)['likes'].min(),
                     gGBvideo.get_group(17)['likes'].min(),
                     gGBvideo.get_group(19)['likes'].min(),
                     gGBvideo.get_group(20)['likes'].min(),
                     gGBvideo.get_group(22)['likes'].min(),
                     gGBvideo.get_group(23)['likes'].min(),
                     gGBvideo.get_group(24)['likes'].min(),
                     gGBvideo.get_group(25)['likes'].min(),
                     gGBvideo.get_group(26)['likes'].min(),
                     gGBvideo.get_group(27)['likes'].min(),
                     gGBvideo.get_group(28)['likes'].min(),
                     gGBvideo.get_group(43)['likes'].min(),
                     gGBvideo.get_group(29)['likes'].min()],
    
    'average_likes':[gGBvideo.get_group(1)['likes'].mean(),
                     gGBvideo.get_group(2)['likes'].mean(),
                     gGBvideo.get_group(10)['likes'].mean(),
                     gGBvideo.get_group(15)['likes'].mean(),
                     gGBvideo.get_group(17)['likes'].mean(),
                     gGBvideo.get_group(19)['likes'].mean(),
                     gGBvideo.get_group(20)['likes'].mean(),
                     gGBvideo.get_group(22)['likes'].mean(),
                     gGBvideo.get_group(23)['likes'].mean(),
                     gGBvideo.get_group(24)['likes'].mean(),
                     gGBvideo.get_group(25)['likes'].mean(),
                     gGBvideo.get_group(26)['likes'].mean(),
                     gGBvideo.get_group(27)['likes'].mean(),
                     gGBvideo.get_group(28)['likes'].mean(),
                     gGBvideo.get_group(43)['likes'].mean(),
                     gGBvideo.get_group(29)['likes'].mean()
                   ]}

dfLikesGB = pd.DataFrame(data_likes_gb)

In [50]:
dfLikesGB = pd.merge(dfGBcat1,dfLikesGB,on="id")

print("")
print("DATA LIKES SETIAP KATEGORI DI REGION GREAT BRITAIN")
dfLikesGB


DATA LIKES SETIAP KATEGORI DI REGION GREAT BRITAIN


id                  title  highest_likes  lowest_likes  average_likes
0    1       Film & Animation         461135             0   57205.523089
1    2       Autos & Vehicles         114841            61   24608.506944
2   10                  Music        5613827             0  272138.508943
3   15         Pets & Animals         187524           147   28629.913858
4   17                 Sports        1291139             0   39270.313057
5   19        Travel & Events          23791            54    9674.447917
6   20                 Gaming        1031695             0   39220.127517
7   22         People & Blogs         625174             0   46273.495899
8   23                 Comedy        1253154            22   78431.168490
9   24          Entertainment        3312868             0   81572.362012
10  25        News & Politics         610526            12   25021.685714
11  26          Howto & Style         341145             5   26639.183091
12  27              Education         383049            94   38440.687090
13  28   Science & Technology         392249             0   61531.127413
14  43                  Shows        2061275           708  271695.733333
15  29  Nonprofits & Activism          44588         34615   41836.900000

#### 3B. Data Disikes Video di Great Britain

In [51]:
# memasukkan data ke dalam list
data_dislikes_gb = {
    
    # a adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Great Britain
    'id' : b,
    'highest_dislikes':[gGBvideo.get_group(1)['dislikes'].max(),
                     gGBvideo.get_group(2)['dislikes'].max(),
                     gGBvideo.get_group(10)['dislikes'].max(),
                     gGBvideo.get_group(15)['dislikes'].max(),
                     gGBvideo.get_group(17)['dislikes'].max(),
                     gGBvideo.get_group(19)['dislikes'].max(),
                     gGBvideo.get_group(20)['dislikes'].max(),
                     gGBvideo.get_group(22)['dislikes'].max(),
                     gGBvideo.get_group(23)['dislikes'].max(),
                     gGBvideo.get_group(24)['dislikes'].max(),
                     gGBvideo.get_group(25)['dislikes'].max(),
                     gGBvideo.get_group(26)['dislikes'].max(),
                     gGBvideo.get_group(27)['dislikes'].max(),
                     gGBvideo.get_group(28)['dislikes'].max(),
                     gGBvideo.get_group(43)['dislikes'].max(),
                     gGBvideo.get_group(29)['dislikes'].max()],
    
    'lowest_dislikes':[gGBvideo.get_group(1)['dislikes'].min(),
                     gGBvideo.get_group(2)['dislikes'].min(),
                     gGBvideo.get_group(10)['dislikes'].min(),
                     gGBvideo.get_group(15)['dislikes'].min(),
                     gGBvideo.get_group(17)['dislikes'].min(),
                     gGBvideo.get_group(19)['dislikes'].min(),
                     gGBvideo.get_group(20)['dislikes'].min(),
                     gGBvideo.get_group(22)['dislikes'].min(),
                     gGBvideo.get_group(23)['dislikes'].min(),
                     gGBvideo.get_group(24)['dislikes'].min(),
                     gGBvideo.get_group(25)['dislikes'].min(),
                     gGBvideo.get_group(26)['dislikes'].min(),
                     gGBvideo.get_group(27)['dislikes'].min(),
                     gGBvideo.get_group(28)['dislikes'].min(),
                     gGBvideo.get_group(43)['dislikes'].min(),
                     gGBvideo.get_group(29)['dislikes'].min()],
    
    'average_dislikes':[gGBvideo.get_group(1)['dislikes'].mean(),
                     gGBvideo.get_group(2)['dislikes'].mean(),
                     gGBvideo.get_group(10)['dislikes'].mean(),
                     gGBvideo.get_group(15)['dislikes'].mean(),
                     gGBvideo.get_group(17)['dislikes'].mean(),
                     gGBvideo.get_group(19)['dislikes'].mean(),
                     gGBvideo.get_group(20)['dislikes'].mean(),
                     gGBvideo.get_group(22)['dislikes'].mean(),
                     gGBvideo.get_group(23)['dislikes'].mean(),
                     gGBvideo.get_group(24)['dislikes'].mean(),
                     gGBvideo.get_group(25)['dislikes'].mean(),
                     gGBvideo.get_group(26)['dislikes'].mean(),
                     gGBvideo.get_group(27)['dislikes'].mean(),
                     gGBvideo.get_group(28)['dislikes'].mean(),
                     gGBvideo.get_group(43)['dislikes'].mean(),
                     gGBvideo.get_group(29)['dislikes'].mean()
                   ]}

dfDislikesGB = pd.DataFrame(data_dislikes_gb)

In [52]:
dfDislikesGB = pd.merge(dfGBcat1,dfDislikesGB,on="id")

print("")
print("DATA DISLIKES SETIAP KATEGORI DI REGION GREAT BRITAIN")
dfDislikesGB


DATA DISLIKES SETIAP KATEGORI DI REGION GREAT BRITAIN


id                  title  highest_dislikes  lowest_dislikes  \
0    1       Film & Animation             37132                0   
1    2       Autos & Vehicles             12647                1   
2   10                  Music            421473                0   
3   15         Pets & Animals              3461                1   
4   17                 Sports            193053                0   
5   19        Travel & Events              2434                0   
6   20                 Gaming             17093                0   
7   22         People & Blogs            258504                0   
8   23                 Comedy             38693                2   
9   24          Entertainment           1944971                0   
10  25        News & Politics             37789                0   
11  26          Howto & Style             13141                3   
12  27              Education              8924                4   
13  28   Science & Technology             19405                0   
14  43                  Shows            545595                5   
15  29  Nonprofits & Activism               361              234   

    average_dislikes  
0        2403.156383  
1        1952.708333  
2       11587.191799  
3         685.505618  
4        5199.032512  
5         361.614583  
6        1288.846756  
7        4102.040328  
8        2533.107768  
9        9656.526852  
10       3085.999184  
11        833.606328  
12       1297.927790  
13       3133.299228  
14      60409.488889  
15        319.800000

#### 4B. Data Comments Video di Great Britain

In [53]:
# memasukkan data ke dalam list
data_comments_gb = {
    
    # b adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Great Britain
    'id' : b,
    'highest_comments':[gGBvideo.get_group(1)['comment_count'].max(),
                     gGBvideo.get_group(2)['comment_count'].max(),
                     gGBvideo.get_group(10)['comment_count'].max(),
                     gGBvideo.get_group(15)['comment_count'].max(),
                     gGBvideo.get_group(17)['comment_count'].max(),
                     gGBvideo.get_group(19)['comment_count'].max(),
                     gGBvideo.get_group(20)['comment_count'].max(),
                     gGBvideo.get_group(22)['comment_count'].max(),
                     gGBvideo.get_group(23)['comment_count'].max(),
                     gGBvideo.get_group(24)['comment_count'].max(),
                     gGBvideo.get_group(25)['comment_count'].max(),
                     gGBvideo.get_group(26)['comment_count'].max(),
                     gGBvideo.get_group(27)['comment_count'].max(),
                     gGBvideo.get_group(28)['comment_count'].max(),
                     gGBvideo.get_group(43)['comment_count'].max(),
                     gGBvideo.get_group(29)['comment_count'].max()],
    
    'lowest_comments':[gGBvideo.get_group(1)['comment_count'].min(),
                     gGBvideo.get_group(2)['comment_count'].min(),
                     gGBvideo.get_group(10)['comment_count'].min(),
                     gGBvideo.get_group(15)['comment_count'].min(),
                     gGBvideo.get_group(17)['comment_count'].min(),
                     gGBvideo.get_group(19)['comment_count'].min(),
                     gGBvideo.get_group(20)['comment_count'].min(),
                     gGBvideo.get_group(22)['comment_count'].min(),
                     gGBvideo.get_group(23)['comment_count'].min(),
                     gGBvideo.get_group(24)['comment_count'].min(),
                     gGBvideo.get_group(25)['comment_count'].min(),
                     gGBvideo.get_group(26)['comment_count'].min(),
                     gGBvideo.get_group(27)['comment_count'].min(),
                     gGBvideo.get_group(28)['comment_count'].min(),
                     gGBvideo.get_group(43)['comment_count'].min(),
                     gGBvideo.get_group(29)['comment_count'].min()],
    
    'average_comments':[gGBvideo.get_group(1)['comment_count'].mean(),
                     gGBvideo.get_group(2)['comment_count'].mean(),
                     gGBvideo.get_group(10)['comment_count'].mean(),
                     gGBvideo.get_group(15)['comment_count'].mean(),
                     gGBvideo.get_group(17)['comment_count'].mean(),
                     gGBvideo.get_group(19)['comment_count'].mean(),
                     gGBvideo.get_group(20)['comment_count'].mean(),
                     gGBvideo.get_group(22)['comment_count'].mean(),
                     gGBvideo.get_group(23)['comment_count'].mean(),
                     gGBvideo.get_group(24)['comment_count'].mean(),
                     gGBvideo.get_group(25)['comment_count'].mean(),
                     gGBvideo.get_group(26)['comment_count'].mean(),
                     gGBvideo.get_group(27)['comment_count'].mean(),
                     gGBvideo.get_group(28)['comment_count'].mean(),
                     gGBvideo.get_group(43)['comment_count'].mean(),
                     gGBvideo.get_group(29)['comment_count'].mean()
                   ]}

dfCommentsGB = pd.DataFrame(data_comments_gb)

In [54]:
dfCommentsGB = pd.merge(dfGBcat1,dfCommentsGB,on="id")

print("")
print("DATA COMMENTS SETIAP KATEGORI DI REGION GREAT BRITAIN")
dfCommentsGB


DATA COMMENTS SETIAP KATEGORI DI REGION GREAT BRITAIN


id                  title  highest_comments  lowest_comments  \
0    1       Film & Animation             50255                0   
1    2       Autos & Vehicles             12083                0   
2   10                  Music           1228655                0   
3   15         Pets & Animals             18532               23   
4   17                 Sports             91663                0   
5   19        Travel & Events              2716                7   
6   20                 Gaming             31803                0   
7   22         People & Blogs            154958                0   
8   23                 Comedy            205692                4   
9   24          Entertainment           1626501                0   
10  25        News & Politics             86221                0   
11  26          Howto & Style            139582                9   
12  27              Education             19451                0   
13  28   Science & Technology             53786                0   
14  43                  Shows            704824               95   
15  29  Nonprofits & Activism              4335             3448   

    average_comments  
0        6049.674428  
1        3507.534722  
2       21303.849062  
3        2630.520599  
4        5273.322496  
5        1231.010417  
6        3449.101790  
7        6395.471292  
8        9145.960613  
9       12812.082420  
10       7662.533878  
11       3815.506224  
12       3581.844639  
13       7778.117761  
14      85253.222222  
15       4047.850000